connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedConfig.java

```diff
@@ -400,6 +424,33 @@ public KeyGenerator getInternalRequestKeyGenerator() {
         }
     }
 
+    private Map<String, Object> topicSettings(String prefix) {
+        Map<String, Object> result = originalsWithPrefix(prefix);
+        if (CONFIG_STORAGE_PREFIX.equals(prefix) && result.containsKey(PARTITIONS_SUFFIX)) {
+            log.warn("Ignoring '{}{}={}' setting, since config topic partitions is always 1", prefix, PARTITIONS_SUFFIX, result.get("partitions"));
+        }
+        Object removedPolicy = result.remove("cleanup.policy");
```

Comment:
It's one more import statement, but I don't see a reason not to get the actual config from: 
`TopicConfig.CLEANUP_POLICY_CONFIG`. I think you'll agree that a variable is better than a string, but just a nit.